### PR TITLE
Update to 5.17.0 and install via npm

### PR DESCRIPTION
**Pull request by jamesmyatt**
Checklist
* [x] Used a fork of the feedstock to propose changes
* [ ] Bumped the build number (if the version is unchanged)
* [x] Reset the build number to `0` (if the version changed)
* [x] [Re-rendered]( https://conda-forge.org/docs/maintainer/updating_pkgs.html#rerendering-feedstocks ) with the latest `conda-smithy` (Use the phrase <code>@<space/>conda-forge-admin, please rerender</code> in a comment in this PR for automated rerendering)
* [x] Ensured the license file is being packaged.

I don't know very much about javascript and npm, but I've taken inspiration from other feedstocks like:

- https://github.com/conda-forge/moment-feedstock/blob/master/recipe/meta.yaml
- https://github.com/conda-forge/svg2png-feedstock/blob/master/recipe/meta.yaml
- https://github.com/conda-forge/gulp-feedstock/blob/master/recipe/meta.yaml

I can't say I'm sure that this is correct, since the test doesn't seem to have worked correctly: https://dev.azure.com/conda-forge/feedstock-builds/_build/results?buildId=248384&view=logs&j=656edd35-690f-5c53-9ba3-09c10d0bea97&t=e5c8ab1d-8ff9-5cae-b332-e15ae582ed2d&l=1142

### CODE REVIEW

**Comment by jamesmyatt** [comment]
@conda-forge-admin, please rerender

File: recipe/meta.yaml [hunk 1]
```diff
@@ -1,29 +1,33 @@
...
   run:
     - nodejs
 
 test:
   commands:
+    - npm list -g vega-cli
     - curl -L https://raw.githubusercontent.com/vega/vega/master/docs/examples/bar-chart.vg.json > test.json
     - vg2pdf test.json > test.pdf
+  requires:
+    - nodejs
```
Comment (xhochy) [inline recipe/meta.yaml:30]: `nodejs` is already part of `run` so not required here.
```suggestion
```

File: recipe/meta.yaml [hunk 2]
```diff
@@ -1,29 +1,33 @@
...
   version: {{ version }}
 
+source:
+  url: https://github.com/vega/vega/archive/v{{ version }}.tar.gz
+  sha256: {{ sha256 }}
+
 build:
   number: 0
-  skip: true  # [win]
+  noarch: generic
```
Comment (xhochy) [inline recipe/meta.yaml:15]: This is isn't possible with NodeJS packages as the executable paths are different on Windows.

**Review by xhochy — CHANGES_REQUESTED** [review]

File: recipe/meta.yaml (hunk 2 above)
Comment (jamesmyatt) [reply recipe/meta.yaml:15]: I suspected as much

**Pull request closed without merging**